src/dispatch/__main__.py

```diff
@@ -70,24 +70,19 @@ def main():
         reference = ARGS.reference
     elif ARGS.task_id:
         reference = ARGS.task_id
-    else:
-        LOGGER.critical("There is something wrong with reference/build_id!")
-        sys.exit(99)
 
-    iterate_over = None
+    plans = ARGS.plans
 
-    if ARGS.plans:
-        iterate_over = ARGS.plans
-    elif ARGS.planfilter:
-        iterate_over = ARGS.planfilter
-    elif ARGS.testfilter:
-        iterate_over = ARGS.testfilter
+    # Exit if multiple plans requested with additional filter(s)
+    if len(plans) > 1 and (ARGS.planfilter or ARGS.testfilter):
```

Comment:
Just curious - what tesar's `plans` argument gets translated into on the tft's side? I see that tft-cli supports `--plan` that then becomes a `name` regex for plans filtering, but you can't have more than 1 plan argument passed this way..

Reply:
It goes to the name regex. More plans are handled by one of the for cycles https://github.com/danmyway/tesar/blob/092069c27295de01343b1759dced0e609108ba0e/src/dispatch/__main__.py#L84
That's why I call the utility a "for cycle simulator" 😅

Reply:
So instead of having `name: plan1` as tft-cli supports it when you pass --plan tesar has `name: plan1&plan2` with name being the test.fmf.name parameter of the tft request ?

Or do you combine it and then pass as `plan-filter=name:plan1 & plan2` ?

Reply:
It's more simple.
It just assigns each plan to an individual payload/request. 
Additionally it does the same for each target.